src/content/docs/en/basics/rendering.mdx

```diff
@@ -0,0 +1,71 @@
+
+<ReadMore>See more about the [`output` setting](/en/reference/configuration-reference/#output) in the configuration reference.</ReadMore>
+
+**By default, your Astro entire astro site will be prerendered** and static HTML pages will be sent to the browser. If you have some pages that require server-rendering, for example a page that checks for cookies and displays personalized content, you can choose to render these individual routes on demand. The rest of your site will be a static site.
+
+For a **highly dynamic app**, set your build output configuration to server-render all your pages by default. Then, you can choose to prerender any individual pages that do not require a server to execute, such as a privacy policy or about page.
+
+## Opting in or out of prerendering
+
+Both of Astro's rendering modes allow you to prerender and to render on demand in the same project. The only difference is which kind of rendering happens by default.
+
+You can set a `prerender` value (`true` or `false`) in your page route to opt in or out of prerendering and override your output's default mode.
+
+<ReadMore>See more about [adding an adapter and using on-demand rendering](/en/guides/on-demand-rendering/) for some or all of your routes.</ReadMore>
+
+### Prerendered pages
+
+**`'static'`:** by default
+
+**`'server'`:** with `export const prerender = true`
+
+Prerendered pages are built ahead of time and ready to send to the browser. The same HTML document is sent to the browser for every visitor, and a full-site rebuild is required to update the contents of the page. This method is also known as **static site generation (SSG)**. 
+
+By default, all Astro projects are configured to be pre-rendered at build time (statically-generated) to provide the most lightweight browser experience. The browser does not need to wait for any HTML to build because the server does not need to generate any pages on demand. Your site is not dependent on the performance of a backend data source, and once built, will remain available to visitors as a static site as long as your server is functioning.
+
+Prerendered pages in Astro can include:
+
+- [Client islands](/en/concepts/islands/) for interactive UI components in an otherwise static, pre-rendered page.
+- [Server islands]() for rendering dynamic or personalized content within a static page shell.
+- [View transitions](/en/guides/view-transitions/) for client-side routing with animation and state persistence across page navigation.
+- [Middleware](/en/guides/middleware/) (with some limitations) to intercept and transform response data from a request.
+
+:::tip
+Astro's default `static` mode is a powerful, modern-feeling choice for content-focused sites with the ability to add individual on-demand routes for dynamic or personalized pages when needed.
+:::
+
+### On-demand rendered routes
+
+**`'static'`:** with `export const prerender = false` and an adapter
+
+**`'server'`:** by default, with an adapter
```

Comment:
nit: layout suggestion (same as above)
I wonder if we shouldn't take the same approach as `Types:`, `Default:`... to avoid the spacing between the two:
```suggestion
<p>

**`'static'`:** with `export const prerender = false` and an adapter<br />
**`'server'`:** by default, with an adapter
</p>
```